mmv1/products/monitoring/api.yaml

```diff
@@ -1084,6 +1084,87 @@ objects:
             The full name of the resource that defines this service.
             Formatted as described in
             https://cloud.google.com/apis/design/resource_names.
+  
+  - !ruby/object:Api::Resource
+    name: GenericService
+    base_url: v3/projects/{{project}}/services
+    create_url: v3/projects/{{project}}/services?serviceId={{service_id}}
+    self_link: "v3/{{name}}"
+    update_verb: :PATCH
+    update_mask: true
+    description: |
+      A Service is a discrete, autonomous, and network-accessible unit,
+      designed to solve an individual concern (Wikipedia). In Cloud Monitoring,
+      a Service acts as the root resource under which operational aspects of
+      the service are accessible
+    references: !ruby/object:Api::Resource::ReferenceLinks
+      guides:
+        'Service Monitoring': 'https://cloud.google.com/monitoring/service-monitoring'
+        'Monitoring API Documentation': 'https://cloud.google.com/monitoring/api/v3/'
+      api: 'https://cloud.google.com/monitoring/api/ref_v3/rest/v3/services'
+    parameters:
+    - !ruby/object:Api::Type::String
+      name: serviceId
+      description: |
+        An optional service ID to use. If not given, the server will generate a
+        service ID.
+      input: true
```

Comment:
This should be required. It makes everything easier if we never let the server generate the service ID for us.

Reply:
We want to have it have parity with the `name: Service` Resource. They are very similar except for how they handle the union field `identifier` (https://cloud.google.com/monitoring/api/ref_v3/rest/v3/services#resource:-service) and `basicService`.

Reply:
Discussed that parity with the `google_monitoring_custom_service` is not as important as a clean API, and we can have a clean API with a new resource.